src/main/java/seedu/duke/Expense.java

```diff
@@ -1,4 +1,30 @@
 package seedu.duke;
 
+import java.util.ArrayList;
+
 public class Expense {
+    private String location;
+    private ArrayList<Person> listOfPersons;
+
+    public String getLocation() {
+        return this.location;
+    }
+
+    public void setLocation(String location) {
+        this.location = location;
+    }
+
+    public void addPersons(ArrayList<Person> newListOfPersons) {
+        listOfPersons.clear();
+        Person personToAdd;
+        Person personToChange;
+        String newName;
+        for (int i = 0; i < newListOfPersons.size(); i++) {
+            personToAdd = newListOfPersons.get(i);
+            personToChange = listOfPersons.get(i);
+            newName = personToAdd.getName();
+            personToChange.setName(newName);
+        }
+    }
```

Comment:
`listOfPersons` cleared but not re-populated here - similarly, line 24 `listOfPersons.get(i)` might throw out of bound exception since list is now empty.

I think you just need to add back each person to `listOfPersons` - correct me if I’m wrong :)

Reply:
Edited this!